Regression test in betxn_test.py failing due to busy LDAP server

**Issue Description**
The regression test dirsrvtests/tests/suites/betxns/betxn_test.py::test_betxn_modrdn_memberof_cache_corruption is failing with the following error:
```
E         ldap.BUSY: {'msgtype': 109, 'msgid': 12, 'result': 51, 'desc': 'Server is busy', 'ctrls': [], 'info': 'target entry busy because of a canceled operation'}

/usr/local/lib64/python3.9/site-packages/ldap/ldapobject.py:128: BUSY
```
This looks to be due to the server still processing the first group.rename() call, which is meant to fail with an ldap.OBJECT_CLASS_VIOLATION exception, as the second group.rename() call is sent in. 
I mitigated this on the test environment with a time.sleep(2), but there should be a better way to fix this. Will keep looking. 

**Package Version and Platform:**
 - Platform: RHEL 9
 - Package and version: Multiple builds, but I found it on 389-ds-base-2.4.5-12.el9_4

**Steps to Reproduce**
Steps to reproduce the behavior:
1. Run the regression test on a RHEL9 test VM. The test is failing on the main branch currently. 
2. The test will fail. 

**Expected results**
The test passes, assuming the build used doesn't cause the test to fail in other ways. 

**Additional context**
I haven't seen this issue documented elsewhere, so I'm opening it here for reference while I investigate further.

--- a/dirsrvtests/tests/suites/betxns/betxn_test.py
+++ b/dirsrvtests/tests/suites/betxns/betxn_test.py
@@ -29,6 +29,16 @@ logging.getLogger(__name__).setLevel(logging.DEBUG)
 log = logging.getLogger(__name__)
 USER_PASSWORD = 'password'
 
+def retry_if_busy(f):
+    for x in range(1000):
+        try:
+            return f()
+        except ldap.BUSY:
+            time.sleep(0.02)
+
+    log.error("Too many retries!")
+    assert False
+
 
 def test_betxt_7bit(topology_st):
     """Test that the 7-bit plugin correctly rejects an invalid update
@@ -240,14 +250,14 @@ def test_betxn_modrdn_memberof_cache_corruption(topology_st):
     if singleTXN:
         # Attempt modrdn that should fail, but the original entry should stay in the cache
         with pytest.raises(ldap.OBJECT_CLASS_VIOLATION):
-            group.rename('cn=group_to_people', newsuperior=peoplebase)
+            retry_if_busy(lambda: group.rename('cn=group_to_people', newsuperior=peoplebase))
 
         # Should fail, but not with NO_SUCH_OBJECT as the original entry should still be in the cache
         with pytest.raises(ldap.OBJECT_CLASS_VIOLATION):
-            group.rename('cn=group_to_people', newsuperior=peoplebase)
+            retry_if_busy(lambda: group.rename('cn=group_to_people', newsuperior=peoplebase))
     else:
-        group.rename('cn=group_to_people', newsuperior=peoplebase)
-        group.rename('cn=group_to_people', newsuperior=peoplebase)
+        retry_if_busy(lambda: group.rename('cn=group_to_people', newsuperior=peoplebase))
+        retry_if_busy(lambda: group.rename('cn=group_to_people', newsuperior=peoplebase))
 
     # Done
     log.info('test_betxn_modrdn_memberof: PASSED')

